### PR TITLE
dmp-4393-amend-get-media-by-id

### PR DESCRIPTION
**Pull request by TomReed**
### JIRA link (if applicable) ###

[DMP-4393](https://tools.hmcts.net/jira/browse/DMP-4393)

### Change description ###

Mappings and models updated to support additional fields as described in LLD and Jira item.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java [hunk 1]
```diff
@@ -271,6 +273,65 @@ void shouldDenyAccess(SecurityRoleEnum role) throws Exception {
...
+
+        // Create hearing and mark it as actual
+        var hearingEntity = databaseStub.createHearing(COURTHOUSE_NAME, COURTROOM_NAME, CASE_NUMBER, HEARING_START_AT.toLocalDateTime());
+        hearingEntity.setHearingIsActual(true);
+        databaseStub.getHearingRepository().save(hearingEntity);
+
+        // Create media and link it to hearing
+        var mediaEntity = createAndSaveMediaEntity(hearingEntity, userAccountEntity, false);
+
+        // Create media linked case with court case (AC1)
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java:292]: Having AC1 in the comment doesn't really mean anything to a dev in the future. Probably worth just removing. Same comment further down.

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java [hunk 2]
```diff
@@ -271,6 +273,65 @@ void shouldDenyAccess(SecurityRoleEnum role) throws Exception {
...
+            .andExpect(jsonPath("$.cases", hasSize(1)))
+            .andExpect(jsonPath("$.cases[0].id").value(hearingEntity.getCourtCase().getId()))
+            .andExpect(jsonPath("$.cases[0].case_number").value(CASE_NUMBER))
+            .andExpect(jsonPath("$.cases[0].source").value("Legacy"))
+            .andExpect(jsonPath("$.cases[0].courthouse.id").value(hearingEntity.getCourtCase().getCourthouse().getId()))
+            .andExpect(jsonPath("$.cases[0].courthouse.display_name").value(COURTHOUSE_NAME));
+    }
+
+    @Test
+    void getMediaById_shouldReturnCaseWithoutCasId_whenMediaHasMigratedCase() throws Exception {
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java:307]: Maybe update the test name? 'MigratedCase' won't mean anything to a future dev so maybe update to how this relates to modernised?

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java [hunk 3]
```diff
@@ -271,6 +273,65 @@ void shouldDenyAccess(SecurityRoleEnum role) throws Exception {
                                     """, jsonString, JSONCompareMode.STRICT);
     }
 
+    @Test
+    void getMediaById_shouldReturnCaseWithCasIdAndActualHearing_whenMediaHasLinkedCase() throws Exception {
```
Comment (chrisbellingham-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java:277]: Suggest including tests cases to verify the newly added `hearings` attributes are returned as expected.

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java (hunk 1 above)
Comment (TomReed) [reply src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java]: agree - have removed

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java (hunk 2 above)
Comment (TomReed) [reply src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java]: now renamed

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java (hunk 3 above)
Comment (TomReed) [reply src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerGetAdminMediasByIdIntTest.java:277]: agree - have added hearings

**Review by chrisbellingham-hmcts — APPROVED** [review]

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**